airbyte_cdk/legacy/sources/declarative/manifest_declarative_source.py

```diff
@@ -341,29 +341,45 @@ def streams(self, config: Mapping[str, Any]) -> List[Union[Stream, AbstractStrea
     def _initialize_cache_for_parent_streams(
         stream_configs: List[Dict[str, Any]],
     ) -> List[Dict[str, Any]]:
+        """Enable caching for parent streams unless explicitly disabled.
+
+        Caching is enabled by default for parent streams to optimize performance when the same
+        parent data is needed by multiple child streams. However, explicit `use_cache: false`
+        settings are respected for streams that cannot use caching (e.g., scroll-based pagination
+        APIs where caching causes duplicate records).
+        """
         parent_streams = set()
 
+        def _should_enable_cache(requester: Dict[str, Any]) -> bool:
+            """Return False only if use_cache is explicitly set to False."""
+            return requester.get("use_cache") is not False
+
+        def _set_cache_if_not_disabled(requester: Dict[str, Any]) -> None:
+            """Set use_cache to True only if not explicitly disabled."""
+            if _should_enable_cache(requester):
+                requester["use_cache"] = True
```

Comment:
Since we only use `_should_enable_cache` once, we probably don't need the separate helper method and can instead just do:
```
def _set_cache_if_not_disabled(requester: Dict[str, Any]) -> None:
  """Set use_cache to True only if not explicitly disabled."""
  if requester.get("use_cache") is not False:
      requester["use_cache"] = True
```